dev/com.ibm.ws.session.cache/src/com/ibm/ws/session/store/cache/SerializationInfoCache.java

```diff
@@ -0,0 +1,468 @@
+            return Arrays.copyOf(bytes, numWritten);
+        }
+    }
+
+    public static final class CharacterInfo implements BuiltinSerializationInfo {
+        public short getIndex() {
+            return CHARACTER;
+        }
+
+        public Class<?> getObjectClass() {
+            return Character.class;
+        }
+
+        public Object bytesToObject(byte[] bytes) {
+            return Character.valueOf(TypeConversion.bytesToChar(bytes, 1));
+        }
+
+        public byte[] objectToBytes(Object object) {
+            byte[] bytes = new byte[] {CHARACTER, 0, 0};
+            TypeConversion.charToBytes(((Character)object).charValue(), bytes, 1);
+            return bytes;
+        }
+    }
+
+    public static final class BooleanInfo implements BuiltinSerializationInfo {
+        private static final byte[] TRUE_BYTES = {BOOLEAN, 1};
+        private static final byte[] FALSE_BYTES = {BOOLEAN, 0};
+
+        public short getIndex() {
+            return BOOLEAN;
+        }
+
+        public Class<?> getObjectClass() {
+            return Boolean.class;
+        }
+
+        public Object bytesToObject(byte[] bytes) {
+            return bytes[1] == 0 ? Boolean.FALSE : Boolean.TRUE;
+        }
+
+        public byte[] objectToBytes(Object object) {
+            return ((Boolean)object).booleanValue() ? TRUE_BYTES : FALSE_BYTES;
+        }
+    }
+
+    public static final class DateInfo implements BuiltinSerializationInfo {
+        public short getIndex() {
+            return DATE;
+        }
+
+        public Class<?> getObjectClass() {
+            return java.util.Date.class;
+        }
+
+        public Object bytesToObject(byte[] bytes) {
+            long l = TypeConversion.varIntBytesToLong(bytes, 1);
+            return new java.util.Date(l);
+        }
+
+        public byte[] objectToBytes(Object object) {
+            long l = ((java.util.Date)object).getTime();
+            byte[] bytes = new byte[] {DATE, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+            int numWritten = TypeConversion.writeLongAsVarIntBytes(l, bytes, 1);
+            return Arrays.copyOf(bytes, numWritten);
+        }
+    }
+
+    public static final class SqlDateInfo implements BuiltinSerializationInfo {
+        public short getIndex() {
+            return SQL_DATE;
+        }
+
+        public Class<?> getObjectClass() {
+            return java.sql.Date.class;
+        }
+
+        public Object bytesToObject(byte[] bytes) {
+            long l = TypeConversion.varIntBytesToLong(bytes, 1);
+            return new java.sql.Date(l);
+        }
+
+        public byte[] objectToBytes(Object object) {
+            long l = ((java.sql.Date)object).getTime();
+            byte[] bytes = new byte[] {SQL_DATE, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+            int numWritten = TypeConversion.writeLongAsVarIntBytes(l, bytes, 1);
+            return Arrays.copyOf(bytes, numWritten);
```

Comment:
Instead of Arrays.copyOf you might as well use System.arrayCopy if it is needed.  Only need to call it numWrite is greater than the array length of course.  This goes for all the calls to Arrays.copyOf.